test/functional/interface_bitcoin_cli.py

```diff
@@ -85,7 +125,7 @@ def run_test(self):
         self.nodes[0].wait_for_cookie_credentials()  # ensure cookie file is available to avoid race condition
         blocks = self.nodes[0].cli('-rpcwait').send_cli('getblockcount')
         self.nodes[0].wait_for_rpc_connection()
-        assert_equal(blocks, BLOCKS)
+        assert_equal(blocks, BLOCKS + 1 if self.is_wallet_compiled() else BLOCKS)
```

Comment:
Could mine the block in the else branch as well, so that subsequent tests don't depend on conditionals?

Reply:
thanks for reviewing -- done